server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -275,6 +288,54 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
         Setting.Property.Deprecated
     );
 
+    /**
+     * Sets the cache size for snapshot repository data: the valid range is within 500Kb  ... 1% of the node heap memory.
+     */
+    public static final Setting<ByteSizeValue> SNAPSHOT_REPOSITORY_DATA_CACHE_THRESHOLD = new Setting<>(
+        SNAPSHOT_REPOSITORY_DATA_CACHET_THRESHOLD_SETTING_NAME,
+        CACHE_MAX_THRESHOLD + "b",
```

Comment:
Just to make sure, we set default to the max value, right? \

Reply:
Hello, @reta!

I read through all our previous comments based on your recent question. While we have discussed min and max values extensively, there wasn’t much conversation about the default value. So, I did some additional research on the topic.
- [`indices.cache.query.size`](https://github.com/elastic/elasticsearch/pull/7161) (now [`indices.queries.cache.size`](https://opensearch.org/docs/latest/search-plugins/caching/request-cache/)): Initially added at 1%, this setting was designed with the premise that “a small cache can be effective.” The configuration`indices.cache.query.expire` allows for setting a time limit after which unused cache entries are removed, as caching the most recently queried items is beneficial in this context.

Since we are caching metadata for snapshots stored in blobs, we chose to use SoftReference as a more suitable safeguard. This implementation suggests that the 1% allocation would not overly burden the system, and additional safeguards make this setting even more manageable. Therefore, setting both the max and default values at 1% seems reasonable.

However, if default and max values are the same, users would only be able to reduce the cache size, making the configuration somewhat limited. If this appears restrictive, we could consider a default slightly below 1% to allow for adjustments both upwards and downwards.

Thank you for bringing up such a thoughtful question! 🎉

Reply:
> However, if having the default and max values the same feels restrictive, as users may only adjust it downwards, we could consider setting the default slightly lower than 1% to allow for more flexibility.

Thank you @inpink , that what I would be inclined to suggest. What would be the good default? I think we could use the following:
 - 500Kb stays as min value
 - set default to `max(500Kb, min(10 x 500Kb, MAX / 2))`: basically, bump limit 10x (to 5Mb) or use 0.5% of the heap as the default

What do you think?

Reply:
Hello, @reta!

Thank you for your excellent suggestion! 😄 I also think setting the default to MAX at `50%` is a great idea. 👍 Based on your input, I’d like to propose using this as the default value:

- `max(500KB, MAX / 2)`

In the case of the default formula you suggested, I think that even if max(1% of heap) is chosen, the default often ends up being `500KB x 10`. I would prefer a default that is also proportional to the heap if memory is not too limited.

If we use the following formula, we could set the default to `50%` of max if `1%` of the heap exceeds `1000KB`.
On the other hand, if `1%` of the heap is less than `1000KB`, we are likely in a very low-memory environment. Therefore, the default could remain as `500KB`, as before.

I’m currently preparing documentation on our discussion to help make these settings more understandable for users. 😁

Thanks for sparking such an interesting discussion on defaults! 👍 It’s great to be working on such a well-tuned feature. Thanks again!